sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2924,6 +2924,50 @@ abstract class JsonSuite
       }
     }
   }
+
+  test("SPARK-35912: nullability with different parse mode -- struct") {
+    val input =
+      """
+        |{
+        |  "c1": {
+        |    "c2": 1
+        |  }
+        |}
+        |""".stripMargin
+    val json = spark.createDataset(spark.sparkContext.parallelize(input :: Nil))(Encoders.STRING)
+
+    val load = (mode: String, schema: StructType) => {
+      spark.read
+        .option("mode", mode)
+        .schema(schema)
+        .json(json)
+    }
+
+    Seq(true, false).foreach { nullable =>
```

Comment:
It looks obvious that the nullability of the outer column doesn't matter. We probably don't need to test it.

Reply:
updated.